src/reseeding.rs

```diff
@@ -59,101 +59,112 @@ impl<R: Rng, Rsdr: Reseeder<R>> ReseedingRng<R, Rsdr> {
     /// generated exceed the threshold.
     /// 
     /// On error, this may delay reseeding or not reseed at all.
-    pub fn reseed_if_necessary(&mut self) {
-        if self.bytes_generated >= self.generation_threshold {
-            let mut err_count = 0;
-            loop {
-                if let Err(e) = self.reseeder.reseed(&mut self.rng) {
-                    // TODO: log?
-                    if e.kind.should_wait() {
-                        // Delay reseeding
-                        let delay = max(self.generation_threshold >> 8, self.bytes_generated);
-                        self.bytes_generated -= delay;
-                        break;
-                    } else if e.kind.should_retry() {
-                        if err_count > 4 {  // arbitrary limit
-                            // TODO: log details & cause?
-                            break;  // give up trying to reseed
-                        }
-                        err_count += 1;
-                        continue;   // immediate retry
-                    } else {
+    #[inline(never)]
+    pub fn reseed(&mut self) {
+        let mut err_count = 0;
+        loop {
+            if let Err(e) = self.reseeder.reseed(&mut self.rng) {
+                // TODO: log?
+                if e.kind.should_wait() {
+                    // Delay reseeding
+                    self.bytes_until_reseed = self.threshold >> 8;
+                    break;
```

Comment:
This sets `bytes_until_reseed` twice. Should it return instead?

I admit that the logic of this function is weird but I guess something like this is useful.

Reply:
Hee, you're right.